xonsh/built_ins.py

```diff
@@ -98,6 +98,44 @@ def expand_path(s):
     return os.path.expanduser(s)
 
 
+braces_regex = re.compile('(.*?){(.*)}(.*)')
```

Comment:
this can be lazy, a function decorated with `@lazyobject` from `xonsh.lazyasd` that returns the regex.

Also i think  that the second `.*` must be the non-greedy one, atm this regex matches weird cases with brackets like:
```
$ braces_regex.fullmatch(r'test test{1,2,3}}test test').groups()
('test test', '1,2,3}', 'test test')
```